addons/sale/tests/test_sale_to_invoice.py

```diff
@@ -1101,3 +1101,52 @@ def test_amount_to_invoice_one_line_multiple_so(self):
 
         self.assertEqual(sale_order_1.amount_to_invoice, -700.0)
         self.assertEqual(sale_order_2.amount_to_invoice, 0.0)
+
+    def test_amount_to_invoice_price_unit_change(self):
+        """
+        We check that the 'amount_to_invoice' relies only on the posted invoice quantity,
+        and is not affected by price changes that occurred during invoice creation.
+        """
+        so = self.env['sale.order'].with_context(tracking_disable=True).create({
+            'partner_id': self.partner_a.id,
+            'partner_invoice_id': self.partner_a.id,
+            'partner_shipping_id': self.partner_a.id,
+            'pricelist_id': self.company_data['default_pricelist'].id,
+        })
+
+        sol_prod_deliver = self.env['sale.order.line'].with_context(tracking_disable=True).create({
+            'product_id': self.company_data['product_order_no'].id,
+            'product_uom_qty': 5,
+            'order_id': so.id,
+            'tax_id': False,
+        })
+
+        so.action_confirm()
+        sol_prod_deliver.write({'qty_delivered': 5.0})
+
+        ctx = {
+            'active_model': 'sale.order',
+            'active_ids': [so.id],
+            'active_id': so.id,
+            'default_journal_id': self.company_data['default_journal_sale'].id,
+        }
```

Comment:
I don't think that the context is relevant?
It would be more explicit to give `sale_order_ids` to the `create` values.